safe-global/web-core
# Fix: don't propose txs w/o signatures

## What it solves

Resolves #1261

## How this PR fixes it
If a transaction is created and immediately executed w/o signatures, don't propose it to the backend.

In this case, we don't show a "View transaction" link in the tx notifications, but the notifications are still grouped by the safeTxHash.

When a smart contract wallet creates a tx, it will still propose a transaction w/o initial signatures.

## How to test it
Case 1:
* Create and immediately execute a tx in a 1/X Safe
* Observe that the tx notifications change each other as the tx is being processed, but there's no deep link in the notifications.
* Once indexed, the tx will appear in the history

Case 2:
* Create and sign a tx
* Execute that tx
* Notifications with a "View transaction" link should appear

Case 3:
* Connect a smart contract wallet, e.g. Safe or Argent
* Create a tx and sign it
* The tx will appear in the queue immediately but with 0 sigs
* Signature will be added when it's indexed

## Analytics changes
None
